python/cudf_polars/cudf_polars/dsl/expressions/rolling.py

```diff
@@ -152,12 +156,166 @@ def do_evaluate(  # noqa: D102
 
 
 class GroupedRollingWindow(Expr):
-    __slots__ = ("options",)
-    _non_child = ("dtype", "options")
+    """
```

Comment:
Naturally shares a lot of the groupy logic with `GroupBy` IR node. A good TODO would be to figure out how much of it can be put into a shared helper.